src/Rules/PHPUnit/NoMissingSpaceInAnnotationRule.php

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PHPUnit;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPUnit\Framework\TestCase;
+use function array_key_exists;
+use function in_array;
+use function preg_match;
+use function preg_split;
+
+/**
+ * @implements Rule<Node>
+ */
+class NoMissingSpaceInAnnotationRule implements Rule
+{
+
+	private const ANNOTATIONS_WITH_PARAMS = [
+		'backupGlobals',
+		'backupStaticAttributes',
+		'covers',
+		'coversDefaultClass',
+		'dataProvider',
+		'depends',
+		'group',
+		'preserveGlobalState',
+		'requires',
+		'testDox',
+		'testWith',
+		'ticket',
+		'uses',
+	];
+
+	/**
+	 * Reflection provider.
+	 *
+	 * @var ReflectionProvider
+	 */
+	private $reflectionProvider;
+
+	public function __construct(ReflectionProvider $reflectionProvider)
+	{
+		$this->reflectionProvider = $reflectionProvider;
+	}
+
+	public function getNodeType(): string
+	{
+		return Node::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node instanceof Node\Stmt\Class_ && !$node instanceof Node\Stmt\ClassMethod) {
+			return [];
+		}
+
+		if ($node instanceof Node\Stmt\Class_) {
+			if ($node->namespacedName !== null && $this->reflectionProvider->getClass($node->namespacedName->toString())->isSubclassOf(TestCase::class) === false) {
```

Comment:
I've looked at it again. These should be two separate rules. The best practice is to have a rule for `InClassNode` - you'll be able to get `$scope->getClassReflection()`. It's gonna work for anonymous classes, and also in some weird situations where the class might actually not exist.

And in case of methods you can use InClassMethodNode and ask for `$scope->getFunction()`. The nice thing about that is that you can then ask for `$scope->getFunction()->getDocComment()` too.

The common part between these two rules can be reused via some helper class that you'll inject into the constructor in both rules.

Thanks for understanding.

Reply:
Thanks, I have updated the code to reflect these changes!